### PR TITLE
Refactor versioned from unversioned cookbooks

### PR DESCRIPTION
**Pull request by jkeiser**
Should make it easier to work with; subclassing to change behavior instead of switch statements.

### CODE REVIEW

File: lib/chef/chef_fs/file_system/chef_server/chef_server_root_dir.rb [hunk 1]
```diff
@@ -141,7 +142,7 @@ def children
             @children ||= begin
               result = [
                 # /cookbooks
-                CookbooksDir.new("cookbooks", self),
+                versioned_cookbooks ? VersionedCookbooksDir.new("cookbooks", self) : CookbooksDir.new("cookbooks", self),
```
Comment (randomcamel) [inline lib/chef/chef_fs/file_system/chef_server/chef_server_root_dir.rb]: Does this mean we can't use versioned and unversioned cookbooks at the same time?

File: lib/chef/chef_fs/file_system/chef_server/cookbooks_dir.rb [hunk 2]
```diff
@@ -29,6 +29,15 @@ class Chef
...
       module ChefServer
+        # cookbooks/
+        #   apache2/
+        #   mysql/
+        # cookbook_artifacts/
+        #   apache2/
+        #     1.0.0/
+        #     1.0.1/
+        #   mysql/
+        #     2.0.5/
```
Comment (randomcamel) [inline lib/chef/chef_fs/file_system/chef_server/cookbooks_dir.rb]: For clarity, the `cookbook_artifacts` should probably have identifiers like the hash strings they'll be in practice.

File: lib/chef/chef_fs/file_system/chef_server/chef_server_root_dir.rb (hunk 1 above)
Comment (jkeiser) [inline lib/chef/chef_fs/file_system/chef_server/chef_server_root_dir.rb]: This is correct. It's repo-wide.

File: lib/chef/chef_fs/file_system/chef_server/cookbooks_dir.rb (hunk 2 above)
Comment (jkeiser) [inline lib/chef/chef_fs/file_system/chef_server/cookbooks_dir.rb]: Agreed.

**Pull request closed without merging**